core/pkg/server/handler.go

```diff
@@ -161,7 +161,37 @@ func (h *Handler) Do(inChan <-chan *spb.Record) {
 	h.logger.Info("handler: started", "stream_id", h.settings.RunId)
 	for record := range inChan {
 		h.logger.Debug("handle: got a message", "record_type", record.RecordType, "stream_id", h.settings.RunId)
+
 		h.handleRecord(record)
+
+		switch record.RecordType.(type) {
+		case *spb.Record_Alert:
+			h.fwdRecord(record)
+		case *spb.Record_Artifact:
+			h.fwdRecord(record)
+		case *spb.Record_Config:
+			h.fwdRecord(record)
+		case *spb.Record_Files:
+			h.fwdRecord(record)
+		case *spb.Record_Output:
+			h.fwdRecord(record)
+		case *spb.Record_OutputRaw:
+			h.fwdRecord(record)
+		case *spb.Record_Preempting:
+			h.fwdRecord(record)
+		case *spb.Record_Stats:
+			h.fwdRecord(record)
+		case *spb.Record_Telemetry:
+			h.fwdRecord(record)
+		case *spb.Record_UseArtifact:
+			h.fwdRecord(record)
+
+		case *spb.Record_Final:
+		case *spb.Record_Footer:
+		case *spb.Record_NoopLinkArtifact:
+		default:
+			// No-op.
```

Comment:
should we error out here as well , i know it is redundant but just to be on the safe side

Reply:
what do you mean? these aren't error cases, we just don't forward these records

Reply:
i meant in case of default it means that this is unknown record
(i know we already check it earlier, but just being extra paranoid)

Reply:
or a record that needs to be forwarded that we miss

Reply:
that's maybe a better reason to add this panic

Reply:
this PR doesn't change all records to use the new path; the default case is for records that I haven't updated and that still get forwarded by `handleRecord` (e.g. exit, header, history, metric, etc)

Reply:
ah i see... got ya!